src/main/java/reposense/authorship/model/AuthorshipSummary.java

```diff
@@ -27,10 +31,29 @@ public void addAuthorContributionCount(Author author) {
         authorFinalContributionMap.put(author, authorFinalContributionMap.get(author) + 1);
     }
 
+    /**
+     * Increments the contribution count of a given filetype for {@code author} .
+     */
+    public void addAuthorFileTypeContributionCount(Author author, String fileType) {
+        TreeMap<String, Integer> fileTypeContributionMap = authorFileTypeContributionMap.get(author);
+
+        if (fileTypeContributionMap.containsKey(fileType)) {
+            fileTypeContributionMap.put(fileType, fileTypeContributionMap.get(fileType) +  1);
+        } else {
+            fileTypeContributionMap.put(fileType, 1);
+        }
+
+        authorFileTypeContributionMap.put(author, fileTypeContributionMap);
```

Comment:
is it necessary to bind back to the author if `authorFileTypeContributionMap.get(author);` works in the first place?